### PR TITLE
Leaderboards UI | Part 2

### PR DESCRIPTION
**Pull request by mlytvynchuk**
Created a custom UI for leaderboards.

### CODE REVIEW

File: app/src/main/java/com/google/droidjump/LeaderboardScoresFragment.java [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+        scoresView.setAdapter(adapter);
+        fetchScores(timeSpan, collection);
+        Switch friendsSwitch = rootView.findViewById(R.id.friends_switch);
+        friendsSwitch.setOnClickListener(ignored -> {
+            if (friendsSwitch.isChecked()) {
+                collection = LeaderboardVariant.COLLECTION_FRIENDS;
+            } else {
+                collection = LeaderboardVariant.COLLECTION_PUBLIC;
+            }
+            Log.d(getClass().toString(), String.valueOf(collection));
```
Comment (redls) [inline app/src/main/java/com/google/droidjump/LeaderboardScoresFragment.java:84]: I am not a fan of having getClass().toString(). Can we have a variable with the name of the class and use that.

Also I am not surer if this log has much value here.

File: app/src/main/java/com/google/droidjump/LeaderboardScoresFragment.java [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+        return rootView;
+    }
+
+    private void fetchScores(int timeSpan, int collection) {
+        Games.getLeaderboardsClient(activity, activity.getSavedSignedInAccount())
+                .loadPlayerCenteredScores(
+                        leaderboard.getLeaderboardId(), timeSpan,
+                        collection, GameConstants.SCORES_PER_PAGE, /* forceReload= */ false)
+                .continueWithTask(task -> {
+                    LeaderboardScoreBuffer scoreBuffer = task.getResult().get().getScores();
```
Comment (redls) [inline app/src/main/java/com/google/droidjump/LeaderboardScoresFragment.java:98]: what if it is not successsful? a in the call fails due to network error?

File: app/src/main/res/layout/leaderboards_item.xml [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+
+
+    <TextView
+        android:id="@+id/leaderboard_item_name"
+        style="@style/RegularText"
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content"
+        android:layout_marginStart="@dimen/fab_margin"
+        android:layout_marginLeft="@dimen/fab_margin"
+        android:text="@string/leaderboards"
```
[the remark this answers is not in the document]
Comment (mlytvynchuk) [reply app/src/main/res/layout/leaderboards_item.xml:52]: Do not catch what you mean. Could you explain?

File: app/src/main/java/com/google/droidjump/LeaderboardScoresFragment.java (hunk 1 above)
Comment (mlytvynchuk) [reply app/src/main/java/com/google/droidjump/LeaderboardScoresFragment.java:84]: Removed.

File: app/src/main/java/com/google/droidjump/LeaderboardScoresFragment.java (hunk 2 above)
Comment (mlytvynchuk) [reply app/src/main/java/com/google/droidjump/LeaderboardScoresFragment.java:98]: Discussed on GVC.

File: app/src/main/java/com/google/droidjump/leaderboards_data/LeaderboardsScoresAdapter.java [hunk 4]
```diff
@@ -59,15 +64,15 @@ public int getItemCount() {
     }
 
     /**
-     * Holds players data.
+     * Gives access to leaderboards_score_item layout and inserts scores data to it.
      */
-    public class PlayersHolder extends RecyclerView.ViewHolder {
+    public class ScoresHolder extends RecyclerView.ViewHolder {
```
Comment (redls) [inline app/src/main/java/com/google/droidjump/leaderboards_data/LeaderboardsScoresAdapter.java:69]: why is this public?

File: app/src/main/java/com/google/droidjump/leaderboards_data/LeaderboardsAdapter.java [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+
+    @Override
+    public int getItemCount() {
+        return items.size();
+    }
+
+    /**
+     * Gives access to leaderboards_item layout and inserts leaderboards data to it.
+     */
+    public class LeaderboardsHolder extends RecyclerView.ViewHolder {
```
Comment (redls) [inline app/src/main/java/com/google/droidjump/leaderboards_data/LeaderboardsAdapter.java:72]: same as before: why is this public?

**Review by redls — APPROVED** [review]

**Pull request merged**